app/tasks/maintenance_tasks/task.rb

```diff
@@ -84,8 +83,7 @@ def collection
     # @raise [NotImplementedError] with a message advising subclasses to
     #   implement an override for this method.
     def process(_item)
-      raise NotImplementedError,
-        "#{self.class.name} must implement `process`."
+      raise "#{self.class.name} must implement `process`."
```

Comment:
This is not a `RuntimeError`, so it seems wrong to raise that class to an error that is really a programming error. Would you raise an `RuntimeError` for a `SyntaxError`?

If we know that `NotImplementedError` can happen in our syste, why we don't rescue it in the `Job#process`?